Modules/Registration/PDEDeformable/include/itkESMDemonsRegistrationFunction.hxx

```diff
@@ -324,7 +324,7 @@ ESMDemonsRegistrationFunction<TFixedImage, TMovingImage, TDisplacementField>::Co
   }
   else if (this->m_UseGradientType == GradientEnum::MappedMoving)
   {
-    PointType mappedPoint;
+    PointType mappedPoint{};
     this->GetFixedImage()->TransformIndexToPhysicalPoint(index, mappedPoint);
```

Comment:
Thanks Jon. I see, you encountered the following warning:

    warning: iteration 2 invokes undefined behavior [-Waggressive-loop-optimizations]
    331 |       mappedPoint[j] += it.GetCenterPixel()[j];

I don't entirely understand why it claims that there is undefined behavior here. The mappedPoint is entirely set by the `TransformIndexToPhysicalPoint(index, mappedPoint)` call. (The second parameter of TransformIndexToPhysicalPoint is an "out" parameter.) Is it indeed a misplaced warning?

Reply:
I don't know Niels. Feel free to investigate.

Reply:
Do you get those warings from a regular GCC build? I see, https://open.cdash.org/viewBuildError.php?type=1&buildid=10154210 says it is from "Ubuntu-22.04-gcc11.2-Rel-Python"

Reply:
I have not built it locally, and the only reference I have is the dashboard.

Reply:
Thanks Jon! I'll see if I can reproduce it with a little example, at the online compiler explorer, something like https://godbolt.org/z/YnqPM7Ee7 (still don't see the warning, but I'll give it another try)